homeassistant/components/pilight.py

```diff
@@ -79,6 +86,8 @@ def send_code(call):
 
         try:
             pilight_client.send_code(message_data)
+            if send_delay > 0.0:
+                time.sleep(send_delay)
```

Comment:
The question is, how can I know how many calls are in the queue and since when to calculate the delay correctly? So my solution is, to manage a queue of actions within the service call. The first call runs directly, any subsequent service call will be put in the action queue. The actions itself schedule then the next action into the HA mainloop at the correct time until all queued actions are completed.